### ISSUE TITLE
Оригинальные названия файлов

### ISSUE THREAD

**Issue by VanTuz5** [Zaharatot/ExHentaiDownloaderZ_5#1]
Как скачивать файлы с оригинальными именами? Дело в том, что при скачивании не комиксов/манги, а галерей, гораздо удобнее скачивать всё с оригинальными именами.

**Comment by Zaharatot** [comment]
На данный момент такой возможности нет. На самом деле - вы, наверное, один из первых пользователей данной программы, кроме меня и моих знакомых XD. До этого просто нужды такой не было, чтобы сохранять оригинальные названия - мне гораздо удобнее просматривать скачанное по номерам.

На данный момент, текущая версия лежащая на GitHub обновляться не будет - я наконец перешёл с WinForms на WPF (поняв насколько эта система круче), и потихоньку переписываю данную программу именно под него (для этого было много предпосылок - в интерфейсе слишком много мест, которые приходилось реализовывать через "костыли"). Думаю, в новой версии можно будет добавить настройку сохранения оригинальных имён файлов. Когда я закончу с этим - скину ссылку на новую версию в этот реквест (но, скорого обновления ждать не стоит, т.к. у меня не очень много свободного времени - скорее всего оно будет через несколько недель).

**Comment by VanTuz5** [comment]
Замечательно, буду ждать! И, надеюсь, она будет качать файлы ещё и в оригинальном разрешении.

**Comment by Zaharatot** [comment]
Хм. Странно - не совсем понятно насчёт оригинального разрешения. Программа берёт ссылку на изображение со страницы просмотра конкретного изображения. Т.е. изображение загружается в том же разрешении, в каком оно отображается на сайте.

**Comment by VanTuz5** [comment]
Просто под некоторыми изображениями есть ссылочка "Открыть файл в оригинальном разрешении" и там оно может быть куда больше.

**Comment by Zaharatot** [comment]
Можно пример ссылки на подобную страницу? Просто я в течение многих лет уже сначала качаю, а потом просматриваю - вот похоже и не натыкался на подобное.

**Comment by Zaharatot** [comment]
А, хотя - уже нашёл сам. Угу, в принципе - можно попробовать добавить и такой функционал. Посмотрю по ходу работы.

**Comment by VanTuz5** [comment]
Шикарно) Рад такому отзывчивому разрабу, побольше бы таких)

**Comment by Zaharatot** [comment]
Мда. Ну, собственно - всё как обычно. Похоже, что обещанные 2 недели превратятся в 2 месяца. Тонна планов на новую версию (например - я планировал переделать движок загрузки, и уже закончил это примерно на 60%) и всё застопорилось. За эти 12 дней я смог выкроить в лучшем случае часов 5-6 на работу над новой версией =_=. 

P.S. Чтобы хоть насколько-то сгладить отрицательные новости - функционал получения картинки в максимальном размере, и получение оригинального имени файла я в движок уже встроил.

**Comment by VanTuz5** [comment]
Да не переживайте - как сделаете, так сделаете. Лишь бы работало как следует

**Comment by Zaharatot** [comment]
Короче говоря. Честно скажу - когда будет 6 версия я не знаю. Я откровенно задолбался работать в итерациях: сажусь делать на выходных (когда есть время) ->  смотрю что сделал раньше -> понимаю что нужно многое переделывать -> переделываю...
Постоянно что-то идёт не так, и у меня уже открытие этого проекта вызывает искреннее отвращение.

Поскольку я всё-таки пообещал о добавлении указанного вами функционала, я в итоге сделал так: данная (5) версия программы была в последний раз обновлена - обновил движок парсинга, добавил запрошенные функции (подробнее - на главной странице репозитория написано). А шестую версию я пока замораживаю, до времён когда будет настроение что-то с ней делать.

**Comment by VanTuz5** [comment]
И то радость. А докачивать в случае достижения лимита она умеет?

**Comment by Zaharatot** [comment]
Да, и уже давно. Программу можно закрыть в любой момент, а после перезапуска (и запуска повторной загрузки), она просканирует уже имеющиеся файлы и будет загружать только то что ещё не было загружено. 
Кроме того, при запуске загрузки, вылазит окошко с запросом о том что "найден существующий путь сохранения файлов", и с запросом установки нового пути загрузки. Если ответить "Да" - загрузка начнётся с ноля в новую папку. Если ответить "Нет" - загрузка будет продолжена.

**Comment by VanTuz5** [comment]
Шикарно! Огромное спасибо!

**Comment by VanTuz5** [comment]
Все скачанные файлы битые...

**Comment by Zaharatot** [comment]
Хм, странно. У меня обычно файлы битыми не приходят никогда. Можете скинуть пример битого файла?

**Comment by VanTuz5** [comment]
Я понял в чём дело - лимит кончился. Когда лимит кончается при запущенной проге - она сообщает, что он, собственно, кончился. Но если запустить с уже истекшим лимитом - молчит.

**Comment by Zaharatot** [comment]
Странно. У меня обычно предупреждение выскакивало (если память не изменяет). Вообще, могу точно сказать, что с дефолтными настройками программа позволяет качать в районе 2500-3000 файлов без проблем. Если загружать больше - уже могут быть проблемы с лимитом. При загрузки больших объёмов - желательно тайминги увеличить в настройках. Кстати говоря - одна из причин почему 6 версия так затянулась в том, что я пытался вычислить чёткие тайминги, при которых проблем с лимитом вообще не будет (и меня во время этих изысканий постоянно банили).

**Comment by VanTuz5** [comment]
и снова часть файлов битая и предупреждение не выскочило... Мне кажется, что нужно не париться с таймингами, а просто сделать проверку на исчерпание лимита и совет сменить ip. И да, я качаю оригиналы, потому лимиты кончаются раньше.
Я до этого качал с эксхентая jdownloader-ом и тот на счёт таймингов не парился совершенно, шпарил по пять файлов за раз (или сколько укажешь в настройках).

**Comment by BAXONnt** [comment]
Или я чего недопонял или ещё что. Но при установке галочки "Оригинальные имена файлов" закачка при каждом запуске программы происходит заново (то есть все скачанные файлы перезаписываются). Если галочка снята файлы идут с именами 1,2,3,4..... и при закрытии/запуске программы идёт, как и положено, дальнейшая скачка.

**Comment by Zaharatot** [comment]
> Или я чего недопонял или ещё что.

Да, скорее всего там начинается повторная загрузка из-за проблем с неймингом (т.е. проверка уже загруженных файлов идёт по именно "номерным", а не по оригинальным именам). Править этот момент я уже точно не буду - данной программой я уже года полтора не пользуюсь, и код там (на мой текущий взгляд) - ужасный. По сути, мне нужно полностью переделывать приложение, и я даже пару раз садился за это, но особого времени у меня на это нет. Да и желания тоже. Если хотите - можете самостоятельно внести поправки, взяв мой исходный код.
